x-pack/plugins/security_solution/common/detection_engine/rule_management/api/rules/bulk_actions/request_schema.ts

```diff
@@ -109,10 +113,12 @@ export const BulkActionEditPayloadRuleActions = t.type({
     t.literal(BulkActionEditType.add_rule_actions),
     t.literal(BulkActionEditType.set_rule_actions),
   ]),
-  value: t.type({
-    throttle: ThrottleForBulkActions,
-    actions: t.array(NormalizedRuleAction),
-  }),
+  value: t.intersection([
+    t.partial({ throttle: ThrottleForBulkActions }),
+    t.type({
+      actions: t.array(NormalizedRuleAction),
+    }),
+  ]),
```

Comment:
Sounds good, will make this adjustment! In any case, the action level frequency has priority.

**Here is the complete overview of all the possible cases:**

- **_Both specified_**: we will ignore rule level throttle
- **_Only rule level throttle specified_**: we will transform rule level throttle in action level frequency 
- **_Only action level frequency specified_**: do nothing
- **_Non specified_**: we will set action level frequency to default value: `{ summary: true, throttle: null, notifyWhen: 'onActiveAlert' }`
- **_Rule level throttle and some of action level frequencies are specified_**: we will transform rule level throttle in action level frequency and set all actions with that value
- **_Only some action level frequencies are specified and there is no rule level throttle_**: we will set all action frequencies to default value

Let me know if that makes sense.

Reply:
thanks for writing down this list of priorities

That makes sense. Although on a second thought, I'm reckoning the option
> Rule level throttle and some of action level frequencies are specified: we will transform rule level throttle in action level frequency and set all actions with that value

Might be a bit confusing. There are already some actions with per action frequency, and according to rule 1 (Both specified: we will ignore rule level throttle), per action frequency will take precedence over rule level.
But, it we have at least one of actions w/o frequency, it will affect the rest of actions.